cli/dcoscli/node/main.py

```diff
@@ -65,12 +72,303 @@ def _cmds():
+            'DC/OS backend does not support snapshot capabilities in this '
+            'version. Must be DC/OS >= 1.8')
+
+
+def sizeof_fmt(num, suffix='B'):
+    for unit in ['', 'Ki', 'Mi', 'Gi', 'Ti', 'Pi', 'Ei', 'Zi']:
+        if abs(num) < 1024.0:
+            return "%3.1f%s%s" % (num, unit, suffix)
+        num /= 1024.0
+    return "%.1f%s%s" % (num, 'Yi', suffix)
+
+
+def _get_snapshots_json():
+    '''
+    Get a json with a list of snapshots
+
+    :return: available snapshots on a cluster.
+    :rtype: dict
+    '''
+    return _do_snapshot_request(SNAPSHOT_BASE_URL + '/list/all',
+                                'GET')
+
+
+def _get_snapshots_list():
+    '''
+    Get a list of tuples (snapshot_file_name, file_size), ..
+
+    :return: list of snapshots
+    :rtype: list of tuples
+    '''
+    available_snapshots = []
+    for _, snapshot_files in _get_snapshots_json().items():
+        if snapshot_files is None:
+            continue
+        for snapshot_file_obj in snapshot_files:
+            available_snapshots.append(
+                (os.path.basename(snapshot_file_obj['file_name']),
+                 snapshot_file_obj['file_size']))
+    return available_snapshots
+
+
+def _snapshot_manage(list_snapshots, status, cancel, json):
+    """
+    Manage snapshots
+
+    :param list_snapshots: a list of available snapshots
+    :type  list_snapshots: bool
+    :param status: show snapshot job status
+    :type  status: bool
+    :param cancel: cancel snapshot job
+    :type  cancel: bool
+    :return: process return code
+    :rtype: int
+    """
+    _check_3dt_version()
+    if list_snapshots:
+        if json:
+            emitter.publish(_get_snapshots_json())
+            return 0
+
+        available_snapshots = _get_snapshots_list()
+        if not available_snapshots:
+            emitter.publish("No snapshots")
+            return 0
+        emitter.publish("Available snapshots:")
+        for available_snapshot in sorted(available_snapshots,
+                                         key=lambda t: t[0]):
+            emitter.publish('{} {}'.format(available_snapshot[0],
+                                           sizeof_fmt(available_snapshot[1])))
+        return 0
+    elif status:
+        snapshot_response = _do_snapshot_request(
+            SNAPSHOT_BASE_URL + '/status/all', 'GET')
+        if json:
+            emitter.publish(snapshot_response)
+            return 0
+
+        for host, props in sorted(snapshot_response.items()):
+            emitter.publish(host)
+            for key, value in sorted(props.items()):
+                emitter.publish('  {}: {}'.format(key, value))
+            emitter.publish('\n')
+        return 0
+    elif cancel:
+        snapshot_response = _do_snapshot_request(
+            SNAPSHOT_BASE_URL + '/cancel', 'POST')
+        if json:
+            emitter.publish(snapshot_response)
+            return 0
+
+        emitter.publish(snapshot_response['status'])
```

Comment:
this will fail if `status` doesnt exist. 
` if status in snapshot_response: 
     emitter.publish...
else:
     raise DCOS...```